docs/flow-diagrams/monitoring-import-flow.puml

```diff
@@ -0,0 +1,80 @@
+@startuml
+
+!pragma useVerticalIf on
+title Import System Overview (with Processing Failure Handling)
+
+start
+
+:Import is scheduled or triggered;
+
+:Connect to SFTP and check for files;
+
+if (New files found?) then (yes)
+  repeat
+    :Attempt to stream file to S3;
+
+    if (Upload succeeds?) then (yes)
+      :Mark file as COLLECTED;
+    else (no)
+      :Mark file as UPLOAD_FAILED;
+      :Log error;
+      note right
+        Failed files are retried
+        up to 5 times in future runs
```

Comment:
Not sure if this retry will be the same once @tommaroh implements the CI run.

Reply:
Right, this will change and we will have to update the documentation when it is implemented.